src/lib/krb5/os/locate_kdc.c

```diff
@@ -288,24 +287,47 @@ locate_srv_conf_1(krb5_context context, const krb5_data *realm,
     }
 
     for (i = 0; hostlist[i]; i++) {
-        int port_num;
+        int default_port = 0;
+        int port_num = 0;
         k5_transport this_transport = transport;
         const char *uri_path = NULL;
+        int family = AF_UNSPEC;
 
         hostspec = hostlist[i];
         Tprintf("entry %d is '%s'\n", i, hostspec);
 
-        parse_uri_if_https(hostspec, &this_transport, &hostspec, &uri_path);
-
-        default_port = (this_transport == HTTPS) ? 443 : udpport;
-        code = k5_parse_host_string(hostspec, default_port, &host, &port_num);
-        if (code == 0 && host == NULL)
-            code = EINVAL;
-        if (code)
-            goto cleanup;
+        if (hostspec[0] == '/') {
+            this_transport = UNIXSOCK;
+            family = AF_UNIX;
+            uri_path = hostspec;
+            host = strdup("localhost");
+            if (host == NULL)
+                goto cleanup;
+        } else {
+            parse_uri_if_https(hostspec, &this_transport, &hostspec, &uri_path);
+
+            switch (this_transport) {
+            case TCP_OR_UDP:
+            case TCP:
+            case UDP:
+                default_port = udpport;
+                break;
+            case HTTPS:
+                default_port = 443;
+                break;
+            default:
+                break;
+            }
```

Comment:
Why the expansion of the ternary expression to this switch statement?

Reply:
I think during development we handle also UNX in the switch statement and later removed it again. I find the switch more readable, but feel free to revert back to the ternary.